docker/map.jinja

```diff
@@ -2,13 +2,13 @@
 # vim: ft=jinja
 
 {%- set tplroot = tpldir.split('/')[0] %}
-{%- import_yaml tplroot ~ "/defaults.yaml" as default_settings %}
-{%- import_yaml tplroot ~ "/osfamilymap.yaml" as osfamilymap %}
-{%- import_yaml tplroot ~ "/osfingermap.yaml" as osfingermap %}
-{%- import_yaml tplroot ~ "/osmap.yaml" as osmap %}
-{%- import_yaml tplroot ~ "/codenamemap.yaml" as codenamemap %}
-{%- import_yaml tplroot ~ "/osarchmap.yaml" as osarchmap %}
-{%- import_yaml tplroot ~ "/cpuarchmap.yaml" as cpuarchmap %}
+{%- import_yaml tplroot ~ "/defaults.yaml" as default_settings with context %}
+{%- import_yaml tplroot ~ "/osfamilymap.yaml" as osfamilymap with context %}
+{%- import_yaml tplroot ~ "/osfingermap.yaml" as osfingermap with context %}
+{%- import_yaml tplroot ~ "/osmap.yaml" as osmap with context %}
+{%- import_yaml tplroot ~ "/codenamemap.yaml" as codenamemap with context %}
+{%- import_yaml tplroot ~ "/osarchmap.yaml" as osarchmap with context %}
+{%- import_yaml tplroot ~ "/cpuarchmap.yaml" as cpuarchmap with context %}
```

Comment:
@Lucianovici Grepping through the `.yaml` files, these are the only ones that are using any Jinja for accessing data structures:

```sls
docker/osfamilymap.yaml:{%- if grains.os == 'MacOS' %}
docker/osfamilymap.yaml:    {%- set rootuser = salt['cmd.run']("stat -f '%Su' /dev/console") %}
docker/osfamilymap.yaml:    {%- set rootgroup = salt['cmd.run']("stat -f '%Sg' /dev/console") %}
docker/osfamilymap.yaml:{%- elif grains.os == 'Windows' %}
docker/osfamilymap.yaml:    {%- set rootuser = salt['cmd.run']("id -un") %}
docker/osfamilymap.yaml:{%- endif %}
docker/osfamilymap.yaml:                {%- if 'oscodename' in grains %}
docker/osfamilymap.yaml:        name: deb [arch=amd64] https://download.docker.com/linux/{{ grains.os|lower }} {{ grains.oscodename }} stable
docker/osfamilymap.yaml:                {%- endif %}
docker/osfamilymap.yaml:        key_url: "https://download.docker.com/linux/{{ grains.os|lower }}/gpg"
docker/osfamilymap.yaml:        baseurl: 'https://download.docker.com/linux/{{ grains.os|lower }}/$releasever/$basearch/stable'
docker/osfamilymap.yaml:        gpgkey: 'https://download.docker.com/linux/{{ grains.os|lower }}/gpg'
docker/osfamilymap.yaml:    rootuser: {{ rootuser | d('') }}
docker/osfamilymap.yaml:    rootuser: {{ rootuser | d('') }}
docker/codenamemap.yaml:      - linux-image-extra-{{ grains.kernelrelease }}
docker/osmap.yaml:        baseurl: 'https://download.docker.com/linux/centos/{{ grains.get('osmajorrelease', '') }}/$basearch/stable'
```

Thus, we only need to add `with context` to those particular imports (as was mentioned in the [`postgres-formula` PR](https://github.com/saltstack-formulas/postgres-formula/pull/311#discussion_r633419410)):

```suggestion
{%- import_yaml tplroot ~ "/defaults.yaml" as default_settings %}
{%- import_yaml tplroot ~ "/osfamilymap.yaml" as osfamilymap with context %}
{%- import_yaml tplroot ~ "/osfingermap.yaml" as osfingermap %}
{%- import_yaml tplroot ~ "/osmap.yaml" as osmap with context %}
{%- import_yaml tplroot ~ "/codenamemap.yaml" as codenamemap with context %}
{%- import_yaml tplroot ~ "/osarchmap.yaml" as osarchmap %}
{%- import_yaml tplroot ~ "/cpuarchmap.yaml" as cpuarchmap %}
```

In fact, this is actually a reminder that we really want to move to our new v5 `map.jinja`, which emphasizes moving Jinja out of our YAML files.

Also, #290 has been merged, so if you rebase this PR on top, all the tests should pass.